libminifi/src/ThreadedSchedulingAgent.cpp

```diff
@@ -110,16 +110,15 @@ void ThreadedSchedulingAgent::schedule(std::shared_ptr<core::Processor> processo
     thread_pool_.execute(std::move(functor), future);
   }
   logger_->log_debug("Scheduled thread %d concurrent workers for for process %s", processor->getMaxConcurrentTasks(), processor->getName());
-  processors_running_.insert(processor->getUUIDStr());
-  return;
+  processors_running_.insert(processor->getUUID());
 }
 
 void ThreadedSchedulingAgent::stop() {
   SchedulingAgent::stop();
   std::lock_guard<std::mutex> lock(mutex_);
-  for (const auto& p : processors_running_) {
-    logger_->log_error("SchedulingAgent is stopped before processor was unscheduled: %s", p);
-    thread_pool_.stopTasks(p);
+  for (const auto& processor_id : processors_running_) {
+    logger_->log_error("SchedulingAgent is stopped before processor was unscheduled: %s", processor_id.to_string());
+    thread_pool_.stopTasks(processor_id.to_string());
```

Comment:
yes that seems to be the case, (so processors with greater than one maxConcurrentTasks share the same `task_status_` flag)

Reply:
@arpadboda could you confirm or clarify, as the author/major refactorer of this code?

Reply:
I think it's intentional. 
The IDs are only used to identify the tasks belonging to a given component, the threadpool otherwise doesn't care. 
I wouldn't call it a beautiful design, but works as expected, so I kept it when the code was refactored.